openedx/core/djangoapps/enrollments/serializers.py

```diff
@@ -77,14 +80,48 @@ class CourseEnrollmentSerializer(serializers.ModelSerializer):
     """
     course_details = CourseSerializer(source="course_overview")
     user = serializers.SerializerMethodField('get_username')
+    finished = serializers.SerializerMethodField()
+    grading = serializers.SerializerMethodField()
 
     def get_username(self, model):
         """Retrieves the username from the associated model."""
         return model.username
 
+    def get_finished(self, model):
+        """Retrieve finished course."""
+        course = modulestore().get_course(model.course_id)
+        if course:
+            try:
+                coursegrade = CourseGradeFactory().read(model.user, course).passed
+            except PermissionDenied:
+                return False
+            return coursegrade
+        return False
+
+    def get_grading(self, model):
+        """Retrieve course grade."""
+        course = modulestore().get_course(model.course_id)
+        course_grade = None
+        summary = []
+        current_grade = 0
+        if course:
+            try:
+                course_grade = CourseGradeFactory().read(model.user, course)
+                current_grade = int(course_grade.percent * 100)
+                for section in course_grade.summary.get(u'section_breakdown'):
+                    if section.get(u'prominent'):
+                        summary.append(section)
+            except PermissionDenied:
+                pass
```

Comment:
This won't do anything if it fails permission checks, should we instead return an empty response?
This would wind up looking like a 0% grade instead, but perhaps that's the correct behavior, I'm not sure.

Reply:
The values are initialized at the beginning of the method. If there's a permission issue the response would be:
```python
{
    'current_grade': 0,
    'certificate_eligible': False,
    'summary': [],
}```

Reply:
Right, I'm wondering if it would be better to imply an empty grade like that, or to return an empty list entirely.

Reply:
@kaizoku I think to avoid issues on the frontend it's better to return an empty grade like this. That way the front-end code doesn't have to change if there's a permission error.
Now, I think probably a better solution would be to return a 403. You think that's acceptable?

Reply:
I'm not sure, perhaps the edX reviewer will have some ideas on this?

Reply:
@ormsbee Can you comment on this change? Is this approach feasible?